### PR TITLE
Making Case.run more testable

### PR DESCRIPTION
**Pull request by john-science**
## Description

While looking at a possible duplicate `coveragerc` file, I noticed that the important `Case.run()` method had no code coverage because it was so tremendously long it was untestable. 

So I split the really long method out into something with helper methods, so it could be at least partially testable.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/cases/case.py [hunk 1]
```diff
@@ -341,14 +342,41 @@ def run(self):
...
+            else:
+                o.operate()
+
+        # if in the settings, report the coverage and profiling
+        Case._endCoverage(cov)
+        Case._endProfiling(profiler)
+
+    def _startCoverage(self):
+        """Helper to the Case.run(): spin up the code coverage tooling,
+        iff the CaseSettings file says to.
```
Comment (opotowsky) [inline armi/cases/case.py:373]: typo: iff

Comment (opotowsky) [reply armi/cases/case.py:373]: Or do you mean if and only if? :-)

This applies to all the other new cov/profiling docstrings too

**Review by opotowsky — COMMENTED** [review]
This looks straightforward. No changes, just breaking of chunks of code and testing them. Good work! 

I'm confused as to why the coverage is failing and the previous runs had a  drop of 5%....do you know what's going on there?

File: armi/cases/case.py (hunk 1 above)
Comment (john-science) [reply armi/cases/case.py:373]: iff = "if and only if"

But if that's not clear, I can change it.

Comment (opotowsky) [reply armi/cases/case.py:373]: It's fine!

**Review by opotowsky — APPROVED** [review]
Oh! No testing coverage=True then!

**Pull request merged**